### PR TITLE
Added Quick start guide to the Javadoc.

### LINKED ISSUE #1512: Overview of Realm Classes
Per this comment in https://github.com/realm/realm-java/pull/1265#issuecomment-144560453

### PR DESCRIPTION
**Pull request by cmelchior**
Fixes #1512 
Fixes https://github.com/realm/realm.io/pull/1717

TODO
- [x] Replace image with proper one.

This adds a Quick start guide to the Javadocs that introduces the the primary API classes. 
image

What do you think @donnfelker @timanglade ?

### CODE REVIEW

**Comment by donnfelker** [comment]
I like it, and if we can add a image in there with the `doc-files` then it will really pop off nicely. 

My main concern though is that you have 908 unread emails. :)

**Comment by cmelchior** [comment]
The image is just below what you can see on the the screen. A good size for images seems to be around 650 px wide. At least on my retina screen + it provides some buffer for people with smaller screens. Also up to 917 now thanks to the fire hose that is GitHub notifications :)

File: realm/realm-library/build.gradle [hunk 1]
```diff
@@ -67,10 +67,17 @@ task javadoc(type: Javadoc) {
...
         charSet = 'UTF-8'
         locale = 'en_US'
+        overview = 'src/overview.html'
     }
     exclude '**/internal/**'
     exclude '**/BuildConfig.java'
     exclude '**/R.java'
+    doLast {
+        copy {
+            from "src/realm-architecture.png"
```
Comment (bmunkholm) [inline realm/realm-library/build.gradle]: I would call it something different though. realm-class-overview.png maybe

**Comment by timanglade** [comment]
Cool! Can we put the diagram above the quick text? I think something visual would really help attract people’s attention.

**Comment by cmelchior** [comment]
Yes, we can move the diagram before the text, no problem.

**Comment by timanglade** [comment]
See diagram discussion in progress here: https://github.com/realm/realm.io/pull/1717#issuecomment-181517149 We can probably reuse the same diagram layout across iOS & Android. Please note that (in my mind at least :D) the diagram is not meant to be a traditional class inheritance structure diagram, but rather a logical Realm diagram.

**Comment by cmelchior** [comment]
Ready for review @realm/java

**Comment by beeender** [comment]
Nice! :+1:

**Comment by stk1m1** [comment]
:+1: Nice diagram!

**Comment by kneth** [comment]
:+1:

**Pull request merged**